### PR TITLE
[Test] SO_REUSEPORT

### PR DESCRIPTION
**Pull request by pkositsyn**
**Description:**
Test PR to show how SO_REUSEPORT might work. All detailes are in the issue

**Link to tracking Issue:** #2670

### CODE REVIEW

File: config/confignet/confignet.go [hunk 1]
```diff
@@ -30,14 +34,28 @@ type NetAddr struct {
...
-	return net.Listen(na.Transport, na.Endpoint)
+	l := net.ListenConfig{
+		Control: control,
+	}
+	return l.Listen(context.Background(), na.Transport, na.Endpoint)
+}
+
+func control(_ string, _ string, c syscall.RawConn) error {
+	return c.Control(func(fd uintptr) {
+		unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEPORT, 1)
```
Comment (jpkrohling) [inline config/confignet/confignet.go:57]: I would assume we need a build tag + alternative implementation for Windows?

File: config/confighttp/confighttp.go [hunk 2]
```diff
@@ -103,8 +104,7 @@ func (interceptor *headerRoundTripper) RoundTrip(req *http.Request) (*http.Respo
 }
 
 type HTTPServerSettings struct {
-	// Endpoint configures the listening address for the server.
-	Endpoint string `mapstructure:"endpoint"`
```
Comment (jpkrohling) [inline config/confighttp/confighttp.go:107]: This is probably breaking quite a few components in the contrib repo, right?

File: config/confignet/confignet.go (hunk 1 above)
Comment (pkositsyn) [reply config/confignet/confignet.go:57]: Sure, everything is mentioned in the issue. This PR is just to reproduce my example in the issue

File: config/confighttp/confighttp.go (hunk 2 above)
Comment (pkositsyn) [reply config/confighttp/confighttp.go:107]: Yes, need to synchronize changes

File: config/confignet/confignet.go (hunk 1 above)
Comment (pkositsyn) [reply config/confignet/confignet.go:57]: I believe there is no alternative for windows. Probably, need to leave generic implementation for other systems

**Pull request closed without merging**